core/src/main/java/com/amplifyframework/logging/BroadcastLogger.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.logging;
+
+import android.annotation.SuppressLint;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * An implementation of {@link Logger} that emits logs to all loggers.
+ */
+public final class BroadcastLogger implements Logger {
+    // List of loggers to emit logs to.
+    private final List<Logger> delegates;
+
+    /**
+     * Creates a new BroadcastLogger.
+     * @param delegates the list of loggers to emit logs to
+     */
+    public BroadcastLogger(@Nullable List<Logger> delegates) {
+        this.delegates = new ArrayList<>();
+        if (delegates != null) {
+            this.delegates.addAll(delegates);
+        }
```

Comment:
Hm. Actually, you could refuse to create a `BroadcastLogger` if `delegates.size() < 1`. That'd remove the need for your `if (delegates.isEmpty())` checks, anyway.

Reply:
Good point. Also, delegates.size() should never be less than 1 since a logger from the defaultPlugin is always added.